lib/logstash/inputs/dynamodb.rb

```diff
@@ -0,0 +1,359 @@
+  config :streams_endpoint, :validate => :string
+
+  # AWS credentials access key.
+  config :aws_access_key_id, :validate => :string, :default => ""
+
+  # AWS credentials secret access key.
+  config :aws_secret_access_key, :validate => :string, :default => ""
+
+  # A flag to indicate whether or not the plugin should scan the entire table before streaming new records.
+  # Streams will only push records that are less than 24 hours old, so in order to get the entire table
+  # an initial scan must be done.
+  config :perform_scan, :validate => :boolean, :default => true
+
+  # A string that uniquely identifies the KCL checkpointer name and cloudwatch metrics name.
+  # This is used when one worker leaves a shard so that another worker knows where to start again.
+  config :checkpointer, :validate => :string, :default => "logstash_input_dynamodb_cptr"
+
+  # Option to publish metrics to Cloudwatch using the checkpointer name.
+  config :publish_metrics, :validate => :boolean, :default => false
+
+  # Option to not automatically stream new data into logstash from DynamoDB streams.
+  config :perform_stream, :validate => :boolean, :default => true
+
+  # Number of read operations per second to perform when scanning the specified table.
+  config :read_ops, :validate => :number, :default => 1
+
+  # Number of threads to use when scanning the specified table
+  config :number_of_scan_threads, :validate => :number, :default => 1
+
+
+  # Number of threads to write to the logstash queue when scanning the table
+  config :number_of_write_threads, :validate => :number, :default => 1
+
+  # Configuation for how the logs will be transferred.
+  # plain is simply pass the message along without editing it.
+  # dynamodb will return just the data specified in the view_format in dynamodb format.
+    # For more information see: docs.aws.amazon.com/amazondynamodb/latest/developerguide/DataFormat.html
+  # json_drop_binary will return just the data specified in the view_format in JSON while not including any binary values that were present.
+  # json_binary_as_text will return just the data specified in the view_format in JSON while including binary values as base64-encoded text.
+  config :log_format, :validate => [LF_PLAIN, LF_DYNAMODB, LF_JSON_NO_BIN, LF_JSON_BIN_AS_TEXT], :default => "plain"
+
+  public
+  def build_credentials
+    if not @aws_access_key_id.to_s.empty? and not @aws_secret_access_key.to_s.empty?
+      @logger.info("Using static credentials: " + @aws_access_key_id + ", " + @aws_secret_access_key)
+      basic = AmazonCredentials::BasicAWSCredentials.new(@aws_access_key_id, @aws_secret_access_key)
+      return AmazonCredentials::StaticCredentialsProvider.new(basic)
+    else
+      @logger.info("Using default provider chain")
+      return AmazonCredentials::DefaultAWSCredentialsProviderChain.new()
+    end # if neither aws access keys
+  end # def build_credentials
+
+  public
+  def register
+    if not LogStash::Environment.jruby?
```

Comment:
We can safely remove this check, for now ```Logstash```  works with JRuby and there are not major efforts to keep compatibility with CRuby. However in the past there were efforts by @ph to make Logstash fully MRI compatible, so this is why all this checks come to place.